File: classes/Platform.lua
local RectangularObject = require("classes.RectangularObject")

local default_width = 6 * U
local default_height = 0.5 * U
local default_color = { 255/255, 97/255, 158/255 }

local Platform = RectangularObject:extends()

function Platform:init(world, px, py --[[, width, height, color, type]])
    -- function RectangularObject:init(
    --     world,
    --     px, py,
    --     width, height,
    --     color,
    --     type
    -- )
    Platform:super("init",
        world,
        px, py,
        default_width, default_height,
        default_color
    -- width or default_width, height or default_height,
    -- color or default_color,
    -- type or "static"
    )
end

Platform.generate_platforms = function(world, map)
    local platforms = {}
    for i, coords in ipairs(map) do
        table.insert(
            platforms,
            Platform(
                world,
                coords[1],
                coords[2]
            )
        )
    end

    -- local tablasdfae = {}
    -- tablasdfae.body = {}
    -- tablasdfae.body.getX = function()
    --     return "gotYa"
    -- end

    -- table.insert(platforms, tablasdfae)

    return platforms
end

return Platform

File: classes/in_Java/Platform.java
/**
 * What: Child of RectangularObject.
 * 
 * Why: We need separate drawing functions for rectangle and circle.
 * 
 * Origin: Self made
 */
public class Platform extends RectangularObject {
    public final static int DEFAULT_WIDTH = 100;
    public final static int DEFAULT_HEIGHT = 20;
    public final static Color DEFAULT_COLOR = new Color(1, 1, 1);
    public final static BodyType DEFAULT_TYPE = BodyType.STATIC;

    public Platform(World world, int px, int py, int width, int height, Color color, BodyType type) {
        super(world, px, py, width, height, color, type);
    }

    public Platform(World world, int px, int py) {
        this(world, px, py, DEFAULT_WIDTH, DEFAULT_HEIGHT, DEFAULT_COLOR, DEFAULT_TYPE);
    }

    public void draw() {
        System.out.print("Platform!");
        super.draw();
    }

}
